samples/apps/logging/js/app.json

```diff
@@ -29,6 +29,30 @@
         "openapi": {}
       }
     },
+    "/log/private/historical": {
+      "get": {
+        "js_module": "logging.js",
+        "js_function": "get_historical",
+        "forwarding_required": "never",
+        "execute_outside_consensus": "never",
+        "authn_policies": ["jwt", "user_cert"],
+        "historical": true,
+        "readonly": true,
+        "openapi": {}
+      }
+    },
+    "/log/private/historical_receipt": {
+      "get": {
+        "js_module": "logging.js",
+        "js_function": "get_historical_with_receipt",
+        "forwarding_required": "never",
+        "execute_outside_consensus": "never",
+        "authn_policies": ["jwt", "user_cert"],
+        "historical": true,
+        "readonly": true,
```

Comment:
Historical implies readonly, it would be elegant if the config reflected that.

Reply:
`"mode": "readonly|writable|historical"` ? Or were you thinking of something different?

Reply:
Yeah I think that'd be good. Paging @eddyashton for API input, but I think this would have the merit of ruling out historical: true, readonly: false and the validation and error that'd need to go with it.

Reply:
Yes, I think this is a tri-state enum. I like `readonly` and `historical` but not convinced by `writable`. I think `readwrite` is what we've used elsewhere (eg `tx.rw`)? I could also go for `canwrite` or `maywrite` or just `write`, but I think `readwrite` is best.

Reply:
`readwrite` is fine. I'll do the change.

Reply:
@achamayou @eddyashton Done, please have a final look if the change looks good.